weave

```diff
@@ -1197,6 +1202,9 @@ launch() {
                 IPRANGE="${1#*=}"
                 IPRANGE_SPECIFIED=1
                 ;;
+            --hairpin-mode=*)
+                HAIRPIN_MODE="${1#*=}"
```

Comment:
Yes I think the default should be `true`.
Yes we could add `--hairpin-mode` as an option to `weaver` which would apply to the proxy (Docker does its own thing for the plugin).
Yes I think that would be best as a separate PR since I would like the key bug-fix for CNI to go on the 1.9 branch and the `main.go` for `weaver` has changed a lot between 1.9 and 2.0.
So I would suggest just removing the four lines from the `launch)` case in this PR.